File: src/main/java/com/fengsheng/handler/skill_du_ji_a_tos.java
package com.fengsheng.handler;

import com.fengsheng.HumanPlayer;
import com.fengsheng.protos.Role;
import com.fengsheng.skill.ActiveSkill;
import com.fengsheng.skill.SkillId;
import org.apache.log4j.Logger;

import java.util.Arrays;
import java.util.HashSet;

public class skill_du_ji_a_tos extends AbstractProtoHandler<Role.skill_du_ji_a_tos> {
    private static final Logger log = Logger.getLogger(skill_du_ji_a_tos.class);

    @Override
    protected void handle0(HumanPlayer r, Role.skill_du_ji_a_tos pb) {
        ActiveSkill skill = r.findSkill(SkillId.DU_JI);
        if (skill == null) {
            log.error("你没有这个技能");
            return;
        }
        if (new HashSet<>(pb.getTargetPlayerIdsList()).size() != pb.getTargetPlayerIdsCount()) {
            log.error("选择的角色重复" + Arrays.toString(pb.getTargetPlayerIdsList().toArray(new Integer[0])));
            return;
        }
        skill.executeProtocol(r.getGame(), r, pb);
    }
}

File: src/main/kotlin/handler/SkillDuJiATos.kt
package com.fengsheng.handler

import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Role
import com.fengsheng.skill.ActiveSkill
import com.fengsheng.skill.SkillId
import org.apache.logging.log4j.kotlin.logger

class SkillDuJiATos : AbstractProtoHandler<Role.skill_du_ji_a_tos>() {
    override fun handle0(r: HumanPlayer, pb: Role.skill_du_ji_a_tos) {
        val skill = r.findSkill(SkillId.DU_JI) as? ActiveSkill
        if (skill == null) {
            logger.error("你没有这个技能")
            r.sendErrorMessage("你没有这个技能")
            return
        }
        if (HashSet(pb.targetPlayerIdsList).size != pb.targetPlayerIdsCount) {
            logger.error("选择的角色重复${pb.targetPlayerIdsList.joinToString()}")
            r.sendErrorMessage("选择的角色重复${pb.targetPlayerIdsList.joinToString()}")
            return
        }
        skill.executeProtocol(r.game!!, r, pb)
    }
}
